### PR TITLE
Add vesting contract tutorial

### PR DESCRIPTION
**Pull request by michaeljklein**
The [_Vesting Contract_](https://github.com/tqtezos/vesting-contract) represents
an on-chain stream of Tez to a fixed address: the amount of available funds is
updated each second and anyone may vest available funds to the target address.

An example of the Vesting Contract may be found [on Edo2Net](https://better-call.dev/edo2net/KT1VdxSaoxxwBb7s9ZWDCbvR4NN54qfkwE1Q/operations).

### CODE REVIEW

File: docs/token-contracts/vesting.md [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+- `seconds_per_tick := 60`
+- `mutez_per_tick := 1`
+
+Then we can have the following scenarios:
+
+- One minute after origination, anyone may trigger the flush of `1 mutez`.
+- If no one flushes the contract after origination, `10 minutes` later,
+  anyone may trigger the flush of _up to_ `10 Mutez`.
+- If the maximum number is flushed `5 minutes` after origination (`5 Mutez`),
+  anyone may flush up to `10 Mutez` `10 minutes` later
```
Comment (smondet) [inline docs/token-contracts/vesting.md:68]: I would uncapitalize `mutex` for consistency.

File: docs/token-contracts/vesting.md [hunk 2]
```diff
@@ -0,0 +1,346 @@
+---
+id: vesting
+title: Getting started with the Vesting contract
+sidebar_label: Vesting Contract Quick Start
+---
+
+# Introduction
```
Comment (smondet) [inline docs/token-contracts/vesting.md:7]: AFAIU in the articles, we start at level `##` (and the `h1` is left for the global title).

File: docs/token-contracts/vesting.md [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+```
+
+- `(address %target_address)`: The `address` that Tez are vested to
+- `(address %delegate_admin)`: The `address` that may call `setDelegate`
+- `(nat %vested_ticks)`: This should be `0`; it represents the number of ticks
+  already transferred (i.e. none at origination)
+- `(start_time %timestamp)`: When the vesting begins
+- `(nat %seconds_per_tick)`: The time period between allowing more Tez is a `tick`
+- `(nat %mutez_per_tick)`: The number of additional `mutez` that may be vested
+  each tick after `start_time`
```
Comment (smondet) [inline docs/token-contracts/vesting.md:99]: I would consistently punctuate the items of the list.

File: docs/token-contracts/vesting.md [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+
+# Vesting Contract
+
+The Vesting Contract holds Tez and periodically allows some of them to be
+transferred to a fixed address, i.e. "_vested_":
+
+```haskell
+parameter (or (option %setDelegate key_hash)
+              (nat %vest))
+```
```
Comment (smondet) [inline docs/token-contracts/vesting.md:42]: there is no `%default`  `unit` entrypoint?

one must delegate to a baker that doesn't share rewards then!

File: docs/token-contracts/vesting.md [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+
+Replace `TARGET_ADDRESS` and `DELEGATE_ADMIN_ADDRESS` with your two addresses
+from the [Client Setup](/docs/setup/1-tezos-client).
+
+#### Origination
+
+```bash
+$ tezos-client --wait none originate contract VestingContract \
+  transferring 1000 from tz1QS8VYYVDjv7iReBzXeheL6x63A1oATTj8 running \
```
Comment (smondet) [inline docs/token-contracts/vesting.md:118]: why not use the `$DELEGATE_ADMIN_ADDRESS` there?

**Review by smondet — COMMENTED** [review]
LGTM, only minor comments

File: docs/token-contracts/vesting.md (hunk 4 above)
Comment (michaeljklein) [reply docs/token-contracts/vesting.md:42]: I added a note and link to a LIGO wrapper that could be used to add entrypoints

**Review by smondet — APPROVED** [review]

**Pull request merged**